app/models/course/registration.rb

```diff
@@ -0,0 +1,35 @@
+class Course::Registration < ActiveRecord::Base
```

Comment:
I'm starting to wonder if this is necessary.

Isn't this just a Course::CourseUser that is not enabled? This should be a state machine.